### PR TITLE
implement URL and URLSearchParams classes to support commons use cases

### PR DESCRIPTION
**Pull request by natanfeitosa**
## Summary:

Sometimes we need to manipulate url or searchparams, but this is not implemented by default in React Native and to solve this we need to install external dependencies which can make the project a little heavier or can make life a little difficult for new developers, so , thinking like this and being on this side of life, I implemented what was missingin the existing classes

## Changelog:

[GENERAL] [ADDED] - Add implementation of missing methods in URL and URLSearchParams

## Test Plan:

The old test suite for the `URL` class was repurposed with the addition of some new cases, and also a new one for the `URLSearchParams` class was created with the most common use cases.

### CODE REVIEW

**Comment by ndronov** [comment]
Hi, is there any intention to merge this request?
@natanfeitosa @react-native-bot @shergin @mdvacca @sammy-SC

**Comment by natanfeitosa** [comment]
@ndronov I'm waiting for feedback from the maintainers

File: packages/react-native/Libraries/Blob/URL.js [hunk 1]
```diff
@@ -55,74 +55,125 @@ if (
 // Small subset from whatwg-url: https://github.com/jsdom/whatwg-url/tree/master/src
 // The reference code bloat comes from Unicode issues with URLs, so those won't work here.
 export class URLSearchParams {
-  _searchParams: Array<Array<string>> = [];
+  _params: Array<Array<string>> = [];
 
-  constructor(params: any) {
-    if (typeof params === 'object') {
-      Object.keys(params).forEach(key => this.append(key, params[key]));
+  constructor(init?: string | {[key: string]: string | string[] | void}) {
```
Comment (43081j) [inline packages/react-native/Libraries/Blob/URL.js:60]: is this right? the IDL shows this:

```idl
sequence<sequence<USVString>> or record<USVString, USVString> or USVString
```

which suggests it should be:

```ts
init?: Array<Array<string>> | Record<string, string> | string
```

i suspect the sequence of strings is actually a tuple too, like entries: `Array<[string, string]>`

File: packages/react-native/Libraries/Blob/URL.js [hunk 2]
```diff
@@ -55,74 +55,125 @@ if (
...
+
+      for (const value of val) {
+        this.append(key, value);
+      }
+    }
   }
 
-  get(name: string): void {
-    throw new Error('URLSearchParams.get is not implemented');
+  append(key: string, value: string | void): void {
```
Comment (43081j) [inline packages/react-native/Libraries/Blob/URL.js:95]: `value` should always be a string it seems. from typescript's definition:

```ts
 append(name: string, value: string): void;
```

**Comment by 43081j** [comment]
in general i would try match the same interface as in the spec, a good place to see this is in typescript's dom definitions i think:

URLSearchParams:

```ts
/** [MDN Reference](https://developer.mozilla.org/docs/Web/API/URLSearchParams) */
interface URLSearchParams {
    readonly size: number;
    append(name: string, value: string): void;
    delete(name: string, value?: string): void;
    get(name: string): string | null;
    getAll(name: string): string[];
    has(name: string, value?: string): boolean;
    set(name: string, value: string): void;
    sort(): void;
    toString(): string;
    forEach(callbackfn: (value: string, key: string, parent: URLSearchParams) => void, thisArg?: any): void;
}

declare var URLSearchParams: {
    prototype: URLSearchParams;
    new(init?: string[][] | Record<string, string> | string | URLSearchParams): URLSearchParams;
};
```

**Comment by talentlessguy** [comment]
could it be reopened?

**Pull request closed without merging**